sbt-version-policy/src/main/scala/sbtversionpolicy/DependencyCheckReport.scala

```diff
@@ -61,44 +62,120 @@ object DependencyCheckReport {
     def message = "missing dependency"
   }
 
-
+  @deprecated("This method is internal")
   def apply(
     currentModules: Map[(String, String), String],
     previousModules: Map[(String, String), String],
     reconciliations: Seq[(ModuleMatchers, VersionCompatibility)],
     defaultReconciliation: VersionCompatibility
+  ): DependencyCheckReport =
+    apply(
+      Compatibility.BinaryCompatible,
+      currentModules,
+      previousModules,
+      reconciliations,
+      defaultReconciliation
+    )
+
+  private[sbtversionpolicy] def apply(
+    compatibilityIntention: Compatibility,
```

Comment:
We now take into account the compatibility level to check (none, binary compatible, or binary and source compatible)